manifest/pack_pull_git.go

```diff
@@ -188,8 +188,25 @@ func pullGit(ctx context.Context, manifest *Manifest, opts ...pack.PullOption) e
 		WithField("branch", popts.Version()).
 		Infof("git clone")
 
-	if _, err = git.PlainCloneContext(ctx, local, false, copts); err != nil {
-		return fmt.Errorf("could not clone repository: %w", err)
+	if _, err := git.PlainCloneContext(ctx, local, false, copts); err != nil {
+		reps, err := git.PlainOpen(local)
```

Comment:
I haven't looked into the lib, but maybe there is a particular error type returned by `PlainClone` that we can switch on, instead of ignoring all errors?

```
switch {
    case errors.Is(err, git.ErrorDoesNotExists): // I made that error up
        // open, fetch
    case err != nil:
        // fail
}

Reply:
![image](https://github.com/unikraft/kraftkit/assets/46280801/2e25fb54-f01e-4684-917d-cfba04354551)

clone has only a single error case, as far as I can see

Reply:
I can switch on that, sure, no problem

Reply:
I checked, and there is!

- https://github.com/go-git/go-git/blob/3f1cfde283c93f33218c807602e93d47f72f7b90/repository.go#L416
- https://github.com/go-git/go-git/blob/v5.6.1/repository.go#L52

Reply:
(oops, eventual consistency)